src/state_machine.zig

```diff
@@ -1118,1865 +1118,623 @@ const TestContext = struct {
+                        .index = @intCast(u32, @divExact(request.items.len, @sizeOf(Transfer)) - 1),
+                        .result = t.result,
+                    };
+                    try reply.appendSlice(std.mem.asBytes(&result));
+                }
+            },
+            .lookup_account => |a| {
+                assert(operation == null or operation.? == .lookup_accounts);
+                operation = .lookup_accounts;
+
+                try request.appendSlice(std.mem.asBytes(&a.id));
+                if (a.balance) |b| {
+                    var account = accounts.get(a.id).?;
+                    account.debits_pending = b.debits_pending;
+                    account.debits_posted = b.debits_posted;
+                    account.credits_pending = b.credits_pending;
+                    account.credits_posted = b.credits_posted;
+                    try reply.appendSlice(std.mem.asBytes(&account));
+                }
+            },
+            .lookup_transfer => |t| {
+                assert(operation == null or operation.? == .lookup_transfers);
+                operation = .lookup_transfers;
+
+                try request.appendSlice(std.mem.asBytes(&t.id));
+                if (t.exists) {
+                    var transfer = transfers.get(t.id).?;
+                    try reply.appendSlice(std.mem.asBytes(&transfer));
+                }
+            },
+
+            .commit => |commit_operation| {
+                assert(operation == null or operation.? == commit_operation);
+                _ = context.state_machine.prepare(commit_operation, request.items);
+
+                const reply_actual_buffer = try allocator.alignedAlloc(u8, 16, 4096);
+                defer allocator.free(reply_actual_buffer);
+
+                const reply_actual_size = context.state_machine.commit(
+                    0,
+                    1,
+                    commit_operation,
+                    request.items,
+                    reply_actual_buffer,
+                );
+                var reply_actual = reply_actual_buffer[0..reply_actual_size];
 
-        if (vector.result == .ok) {
-            try expectEqual(vector.object, state_machine.get_account(vector.object.id).?.*);
+                if (commit_operation == .lookup_accounts) {
+                    for (std.mem.bytesAsSlice(Account, reply_actual)) |*a| a.timestamp = 0;
+                }
+
+                if (commit_operation == .lookup_transfers) {
+                    for (std.mem.bytesAsSlice(Transfer, reply_actual)) |*t| t.timestamp = 0;
+                }
+
+                // TODO(Zig): Use inline-switch to cast the replies to []Reply(operation), then
+                // change this to a simple "try".
+                testing.expectEqualSlices(u8, reply.items, reply_actual) catch |err| {
+                    print_results("expect", commit_operation, reply.items);
+                    print_results("actual", commit_operation, reply_actual);
```

Comment:
I generally prefer `want/got` to `expect/actual`, though no super strong opinion: https://github.com/ziglang/zig/issues/4437#issuecomment-1328045430

Reply:
I agree that `want/got` are clearer terms, but I'm going to leave it as-is for consistency. (Also, I do like that they align with the same number of letters :stuck_out_tongue:).